common/json.hpp

```diff
@@ -18,23 +18,85 @@
 #include <common/error.hpp>
 #include <common/expected.hpp>
 
-namespace json {
+#include "config.h"
+
+#ifdef MENDER_COMMON_USE_NLOHMANN_JSON
+#include <nlohmann/json.hpp>
+#endif
+
+using namespace std;
```

Comment:
This should be inside the namespace when being used in a header file, otherwise you are doing it globally for everyone who includes this.

Reply:
> otherwise you are doing it globally for everyone who includes this.
Excellent! :grin: